unblob/report.py

```diff
@@ -147,8 +147,9 @@ class FileMagicReport(Report):
 
     @classmethod
     def from_path(cls, path: Path):
-        detected = magic.detect_from_filename(path)
-        return cls(magic=detected.name, mime_type=detected.mime_type)
+        mime_type = magic.Magic(mime=True).from_file(path)
+        full = magic.Magic(keep_going=True).from_file(path)
```

Comment:
I would not initialize Magic every time we need to lookup as that is very expensive.

I would rather create 2 Magic and just reuse it. Even the library has a cache to do so.

for mime based one without keep_going we can use the builtin `from_file`

Reply:
At first the code was using `magic.from_file`, but was changed as you saw it, because it is more explicit.

Will look into caching `Magic()` objects.